packages/lab/src/slider/internal/SliderThumb.tsx

```diff
@@ -1,82 +1,163 @@
+  type HTMLAttributes,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import sliderThumbCss from "./SliderThumb.css";
+import { SliderTooltip } from "./SliderTooltip";
 
 const withBaseName = makePrefixer("saltSliderThumb");
 
-export interface SliderThumbProps extends ComponentPropsWithoutRef<"div"> {
-  index: ThumbIndex;
-  activeThumb: ActiveThumbIndex;
-  setActiveThumb: (index: ActiveThumbIndex) => void;
+interface SliderThumbProps
+  extends Omit<
+    HTMLAttributes<HTMLInputElement>,
+    "onChange" | "defaultValue" | "min" | "max"
+  > {
+  disabled: boolean;
+  format?: (value: number) => number | string;
+  handleInputChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  handlePointerDown: (event: React.PointerEvent<HTMLDivElement>) => void;
+  index?: number;
+  min: number;
+  max: number;
+  name?: string;
+  offsetPercentage?: string;
+  onChange: (event: PointerEvent | React.PointerEvent) => void;
+  step: number;
+  thumbValue: [number, number] | number;
+  trackDragging: boolean;
 }
 
-export function SliderThumb(props: SliderThumbProps): JSX.Element {
-  const { index, activeThumb, setActiveThumb } = props;
+export const SliderThumb = ({
+  "aria-label": ariaLabel,
+  "aria-valuetext": ariaValueText,
+  "aria-describedby": ariaDescribedBy,
+  "aria-labelledby": ariaLabelledBy,
+  disabled,
+  format,
+  handleInputChange,
+  handlePointerDown,
+  index = 0,
+  min,
+  max,
+  name,
+  offsetPercentage,
+  onChange,
+  step,
+  thumbValue,
+  trackDragging,
+  ...rest
+}: SliderThumbProps) => {
+  {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-slider-thumb",
+      css: sliderThumbCss,
+      window: targetWindow,
+    });
 
-  const [focussed, setFocussed] = useState(false);
+    const [isTooltipVisible, setIsTooltipVisible] = useState(false);
+    const [isFocused, setIsFocused] = useState(false);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const id = useId();
 
-  const { min, max, step, value, onChange, ariaLabel } = useSliderContext();
+    useEffect(() => {
+      if (isTooltipVisible) {
+        targetWindow?.addEventListener("keydown", handleKeyDown);
+      }
+      return () => targetWindow?.removeEventListener("keydown", handleKeyDown);
+    }, [targetWindow, isTooltipVisible]);
 
-  const onKeyDown = useKeyDownThumb(min, max, step, value, onChange, index);
+    const handlePointerEnter = () => setIsTooltipVisible(true);
 
-  const thumbValue = value[index];
+    const handlePointerLeave = () => {
+      // Delay hiding the tooltip to enable tooltip
+      // visibility on hover
+      setTimeout(() => {
+        setIsTooltipVisible(false);
+      }, 300);
+    };
 
-  const thumbPosition = getPercentage(min, max, thumbValue as number);
+    const handleFocus = () => {
+      // We add focus to the input to get the keyboard
+      // interactions for the slider out of the box.
+      if (inputRef.current) inputRef.current.focus();
```

Comment:
Is this actually necessary? It seems to call focus on the input when the input is focused?